### ISSUE TITLE
[question] Access environment variable in profile

### ISSUE THREAD

**Issue by shelper** [conan-io/conan#7465]
So i want to setup a android cross building environment  profile that can be shared by the group, and since every one install the  android ndk in different path, i wonder if it is possible to do something like this:
1. user setup ANDROID_NDK environment variable according to their own installation 
2. in the shared cross building profile use something like the cmake does

```
[env]
PATH=[$ENV{ANDROID_NDK}/toolchains/llvm/prebuilt/linux-x86_64/bin]
```
or is there a way to work around for my case?

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by memsharded** [comment]
Hi @shelper 

Right now there is no way to get an environment variable in a profile. I am considering the idea of moving profiles to python files, they already have quite a bunch of logic there, that it is too custom:

- ``include(other_profile)``
- variable replacement

These mechanisms have their own syntax and logic. I think it would be way more flexible, and specially transparent for the users if profiles were python files that would be evaluated. It will also remove complexity from the Conan codebase.

**Comment by memsharded** [comment]
We have merged #9147, that supports jinja2 templating for profiles. This allows not only to read env-vars, or platform configuration, but also to manipulate them, join, conditionals, fors, etc.

So I think this issue is closed by this (will be released next 1.38), please have a look at it and test it when released, feedback very welcome.
